src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -349,6 +349,15 @@ ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
 								  "foreign tables.")));
 	}
 
+#if (PG_VERSION_NUM >= 100000)
+	if (relation->rd_rel->relispartition)
+	{
+		ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
+						errmsg("cannot distribute relation: %s", relationName),
+						errdetail("Distributing partition tables is unsupported.")));
+	}
```

Comment:
This should have a test.  I think this is not sufficient, because it won't catch a partitioned table, just it's partition. Seems we need both for now.